specreduce/tracing.py

```diff
@@ -110,3 +113,164 @@ def __post_init__(self):
+        width_guess = yy_above_half_max / 2.355
+
+        # enforce some (maybe sensible?) rules about trace peak width
+        width_guess = (2 if width_guess < 2
+                       else 25 if width_guess > 25
+                       else width_guess)
+
+        # fit a Gaussian to peak for fall-back answer, but don't use yet
+        g1d_init = models.Gaussian1D(amplitude=np.nanmax(ztot),
+                                     mean=peak_y, stddev=width_guess)
+        offset_init = models.Const1D(np.nanmedian(ztot))
+        profile = g1d_init + offset_init
+
+        fitter = fitting.LevMarLSQFitter()
+        popt_tot = fitter(profile, yy, ztot)
+
+        # restrict fit to window (if one exists)
+        ilum2 = (yy if self.window is None
+                 else yy[np.arange(peak_y - self.window,
+                                   peak_y + self.window, dtype=int)])
+
+        x_bins = np.linspace(0, self.image.shape[self.disp_axis],
+                             self.bins + 1, dtype=int)
+        y_bins = np.tile(np.nan, self.bins)
+
+        for i in range(self.bins):
+            # repeat earlier steps to create gaussian fit for each bin
+            z_i = np.nansum(self.image[ilum2, x_bins[i]:x_bins[i+1]],
+                            axis=self.disp_axis)
+            peak_y_i = ilum2[np.nanargmax(z_i)]
+
+            yy_i_above_half_max = np.sum(z_i > (np.nanmax(z_i) / 2))
+            width_guess_i = yy_i_above_half_max / 2.355
+            width_guess_i = (2 if width_guess_i < 2
+                             else 25 if width_guess_i > 25
+                             else width_guess_i)
+
+            # save initial parameter guesses in case off issues with fits
+            pguess = CompoundModel(
+                '+',
+                models.Gaussian1D(amplitude=np.nanmax(z_i),
+                                  mean=peak_y_i, stddev=width_guess_i),
+                models.Const1D(np.nanmedian(z_i)))
+
+            try:
+                g1d_init_i = models.Gaussian1D(amplitude=np.nanmax(z_i),
+                                               mean=peak_y_i,
+                                               stddev=width_guess_i)
+                offset_init_i = models.Const1D(np.nanmedian(z_i))
+
+                profile_i = g1d_init_i + offset_init_i
+                popt_i = fitter(profile_i, ilum2, z_i)
+
+                # if gaussian fits off chip, then fall back to previous answer
+                if not ilum2.min() <= popt_i.mean_0 <= ilum2.max():
+                    y_bins[i] = popt_tot.mean_0.value
+                else:
+                    y_bins[i] = popt_i.mean_0.value
+                    popt_tot = popt_i
+
+            except RuntimeError:
+                # NOTE: would this happen? if not, pguess and try/except are unneeded
+                warnings.warn(f"TRACE: Fitting bin {i} caused RuntimeError; "
+                              "reverting to initial guess")
+                popt_i = pguess
```

Comment:
if we're not sure if this is needed, I would vote to remove it.  If we are keeping it in, then maybe require passing a flag to fallback on the default?  Otherwise I'm worried it could result in unexpected results (especially if anyone has warnings suppressed).

Reply:
I think this is another case where understanding the original motivation might help.  @jradavenport, do you remember why you had to account for `RuntimeError`s in the loop for creating Gaussian fits to each bin?

Reply:
I _believe_ this was to deal with `curve_fit` failing... not clear to me if its needed still